### DISCUSSION TITLE
Modify the look of units

### DISCUSSION THREAD

**Discussion by BachelorStrange** [stefanhendriks/Dune-II---The-Maker#580] [Q&A] [answered]
Hi! Thank you very much for making available this repository. I have so many nice childhood memories connected to this game and it is really great to have the opportunity to make changes to it now. At the moment, I am just playing around with the code to find out how the parts are related to each other. For this purpose, I would like to change the look of one unit. 

I saw that there are some bmps stored for the UI, but it seems that the units are drawn by the allegro tool. If you have time, I would be happy if you could give a hint on the following questions:

- If I am right and the units are drawn with allegro, then where are the corresponding files stored? 
- Do you think that there is a way to modify the units using allegro? I assume that in this case, I would need a 3rd party tool to create allegro files. Do you know of one?

**Comment by AssociationSirius** [comment, marked as answer]
Do not waste your time !
Unit images are stored in .pak files, it is a tarball type container.
If you modify an image, (center of the sprite, size of the image etc...) you have to reconfigure the sources, which is an nightmare.

Suddenly, (if I dare say) you arrive too early: the easy mod of d2tm is indeed planned, but not current

**Reply by BachelorStrange** [reply]
Ok, thanks. I will have a look at it nevertheless. Just to see how this ordeal looks like :)

**Comment by AssociationSirius** [comment]
To change this I created pull requests 
#508  --> no unreadable data pak without Allegro
#575  --> an example to read data out of code
#572  --> complete example to read all bullets from a text file
Not all of these pull requests have been fully tested

**Reply by BachelorStrange** [reply]
At least modifying existing files seems to work without problems. I found out how to use the grabber tool, which made it possible to load, export and import bmp files. For testing purposes, I modified the mentat bmps, which was possible without changing the code base.

**Comment by stefanhendriks** [comment]
FYI: In d2tm you can unpack the .dat files with grabber and replace them. If you do not change the naming (important!) of those resources (and thus make sure the order does not change) it will work. If you start adding new things, the game will break.
